### PR TITLE
Redirect signed-in users to new personalized home page

### PR DESCRIPTION
**Pull request by varuna13**
Please always provide the Jira Issue your PR is for, as well as test URLs where your change can be observed (before and after):

Jira ID: https://jira.corp.adobe.com/browse/EXLM-2173 

Test URLs:

- Before: https://main--exlm--adobe-experience-league.hlx.live?martech=off
- After: https://feature-exlm-2173--exlm--adobe-experience-league.hlx.live?martech=off

### CODE REVIEW

File: scripts/scripts.js [hunk 1]
```diff
@@ -1246,6 +1248,27 @@ function handleHomePageHashes() {
   return false;
 }
 
+// Function to handle user redirection to personalized home page if signed in
+async function homePageRedirection() {
+  const {lang} = getPathDetails();
+  if (window?.location.pathname === "/" || window?.location.pathname === `/${lang}`) {
+    try {
+      await loadIms();
```
Comment (ruhisingh1) [inline scripts/scripts.js:1256]: we shouldn't always call this method loadIms() when accessing home page.
We should first check if isSignedIn and accessing homepage, then redirect to PHP.

File: scripts/scripts.js [hunk 2]
```diff
@@ -1431,6 +1439,18 @@ if (!window.hlx.DO_NOT_LOAD_PAGE) {
         await window?.adobeIMS?.signIn();
       }
     }
+  } else if (isHomePage(lang)) {
+    try {
+      await loadIms();
+      const { personalizedHomeLink } = getConfig() || {};
+      if (window?.adobeIMS?.isSignedInUser() && personalizedHomeLink) {
+        window.location.replace(`${window.location.origin}${personalizedHomeLink}`);
```
[the remark this answers is not in the document]
Comment (noormd08) [reply scripts/scripts.js:1447]: add return for safer side.

[the remark this answers is not in the document]
Comment (varuna13) [reply scripts/scripts.js:1447]: 'location.replace' will replace the url entry in the browsers session history. So, the users cannot click the “back” button and return to the previous, pre-redirect page.

[the remark this answers is not in the document]
Comment (noormd08) [reply scripts/scripts.js:1447]: Please test this once.

**Review by noormd08 — APPROVED** [review]

File: scripts/scripts.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (varuna13) [reply scripts/scripts.js:1447]: add return for safer side - **Not required**

**Pull request merged**